src/finn/custom_op/fpgadataflow/rtl/dynmvau_rtl.py

```diff
@@ -0,0 +1,348 @@
+                    "{}/input_1.npy".format(code_gen_dir), export_wdt, wnbits
+                )
+                num_w_reps = np.prod(self.get_nodeattr("numInputVectors"))
+                io_dict = {
+                    "inputs": {"in0": inp_0, "in1": inp_1 * num_w_reps},
+                    "outputs": {"out": []},
+                }
+                self.rtlsim_multi_io(sim, io_dict)
+                output = io_dict["outputs"]["out"]
+            else:
+                output = self.rtlsim(sim, inp_0)
+            odt = self.get_output_datatype()
+            target_bits = odt.bitwidth()
+            packed_bits = self.get_outstream_width()
+            out_npy_path = "{}/output.npy".format(code_gen_dir)
+            out_shape = self.get_folded_output_shape()
+            rtlsim_output_to_npy(output, out_npy_path, odt, out_shape, packed_bits, target_bits)
+            # load and reshape output
+            output = np.load(out_npy_path)
+            oshape = self.get_normal_output_shape()
+            output = np.asarray([output], dtype=np.float32).reshape(*oshape)
+            context[node.output[0]] = output
+        else:
+            raise Exception(
+                """Invalid value for attribute exec_mode! Is currently set to: {}
+            has to be set to one of the following value ("cppsim", "rtlsim")""".format(
+                    mode
+                )
+            )
+
+    def get_verilog_top_module_intf_names(self):
+        intf_names = super().get_verilog_top_module_intf_names()
+        mem_mode = self.get_nodeattr("mem_mode")
+        sname = self.hls_sname()
+        if mem_mode == "external":
+            # find the weights_V interface and rename it to in1_V
+            for i, (name, width) in enumerate(intf_names["s_axis"]):
+                if name == "weights_V":
+                    intf_names["s_axis"][i] = ("in1_" + sname, self.get_weightstream_width_padded())
+            # intf_names["s_axis"].append(("in1_" + sname, self.get_weightstream_width_padded()))
+        if mem_mode == "internal_decoupled":
+            # only expose axilite interface if attribute is set
+            runtime_writable = self.get_nodeattr("runtime_writeable_weights") == 1
+            if runtime_writable:
+                intf_names["axilite"] = ["s_axilite"]
+        return intf_names
+
+    def get_folded_input_shape(self, ind=0):
+        mw = self.get_nodeattr("MW")
+        mh = self.get_nodeattr("MH")
+        simd = self.get_nodeattr("SIMD")
+        pe = self.get_nodeattr("PE")
+        sf = mw // simd
+        nf = mh // pe
+        if ind == 0:
+            # calculate shape of input 0
+            vecs = list(self.get_nodeattr("numInputVectors"))
+            folded_input_shape = tuple(vecs + [sf, simd])
+        elif ind == 1:
+            # calculate shape of input 1
+            # TODO: AB: Assume batch size is 1
+            vecs = [1, self.get_nodeattr("MW")]
+            folded_input_shape = tuple(vecs + [nf, pe])
```

Comment:
I'm not sure this is right. e.g. if my dyn matmul has 1x12x32x128 as the B input, I would expect the folded shape here to be [1, 12, 32, 128//PE, PE] whereas the current impl here would give [1, 32, 128//PE, PE]. there needs to be something derived from numInputVectors here - possibly with the last element (N_VECTORS?) dropped